find-route-around-barriers/src/main/java/com/esri/arcgismaps/sample/findroutearoundbarriers/MainActivity.kt

```diff
@@ -0,0 +1,468 @@
+                directionsList.addAll(firstRoute.directionManeuvers)
+            }.onFailure {
+                showError("No route solution. ${it.message}")
+            }
+        }
+    }
+
+    /** Creates a bottom sheet to display a list of direction maneuvers.
+     *  [directions] a list of DirectionManeuver which represents the route
+     */
+    private fun setupBottomSheet(directions: List<DirectionManeuver>) {
+        val bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet).apply {
+            // expand the bottom sheet, and ensure it is displayed on the screen when collapsed
+            state = BottomSheetBehavior.STATE_EXPANDED
+            peekHeight = header.height
+            // animate the arrow when the bottom sheet slides
+            addBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                    imageView.rotation = slideOffset * 180f
+                }
+
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    imageView.rotation = when (newState) {
+                        BottomSheetBehavior.STATE_EXPANDED -> 180f
+                        else -> imageView.rotation
+                    }
+                }
+            })
+        }
+
+        bottomSheet.apply {
+            visibility = View.VISIBLE
+            // expand or collapse the bottom sheet when the header is clicked
+            header.setOnClickListener {
+                bottomSheetBehavior.state = when (bottomSheetBehavior.state) {
+                    BottomSheetBehavior.STATE_COLLAPSED -> BottomSheetBehavior.STATE_EXPANDED
+                    else -> BottomSheetBehavior.STATE_COLLAPSED
+                }
+
+            }
+            // rotate the arrow so it starts off in the correct rotation
+            imageView.rotation = 180f
+
+            directionsLV.apply {
+                // set the adapter for the list view
+                adapter = ArrayAdapter(
+                    this@MainActivity,
+                    android.R.layout.simple_list_item_1,
+                    directions.map { it.directionText }
+                )
+
+                // when the user taps a maneuver, set the viewpoint to that portion of the route
+                onItemClickListener =
+                    AdapterView.OnItemClickListener { _, _, position, _ ->
+                        // remove any graphics that are not the original (blue) route graphic
+                        if (routeOverlay.graphics.size > 1) {
+                            routeOverlay.graphics.removeAt(routeOverlay.graphics.size - 1)
+                        }
+                        // set the viewpoint to the selected maneuver
+                        val geometry = directionsList[position].geometry
+                        geometry?.let { Viewpoint(it.extent, 20.0) }
+                            ?.let { mapView.setViewpoint(it) }
```

Comment:
I think it's better to just use one `let` here because the result of the first `let` will never be null, only if the geometry is null.
```suggestion
                        geometry?.let { mapView.setViewpoint(Viewpoint(it.extent, 20.0)) }
```